pkg/resource/apply.go

```diff
@@ -55,6 +64,9 @@ func (r *Helper) Apply(obj []byte) (ApplyResult, error) {
 	}
 	err = applyOptions.Run()
 	if err != nil {
+		if annotationTooLong(err) {
```

Comment:
I think that I would rather this be something directed by the user rather than fallen back to. In other words, the user should have to explicitly specify in the syncset that they want to use creates and updates rather than applies. That way the user is not surprised by behavior that they would not really have insight into.